### PR TITLE
Remove block_until_ready from the transfer_and_register_cpu_chunk code in TPU Host offload connector

### PR DESCRIPTION
**Pull request by saikat-royc**
# Description

Start with a short description of what the PR does and how this is a change from
the past.

   Eliminates the second synchronous `block_until_ready()` call in the KV
    offload pipeline, enabling fully non-blocking Device-to-Host (D2H)
    transfers. This change decouples the initiation of the data transfer from the
    resource reclamation, ensuring the background thread pool is never
    stalled and the main model loop remains free of XLA synchronization
    bottlenecks.

The rest of the description includes relevant details and context, examples:
- why is this change being made,
- the problem being solved and any relevant context,
- why this is a good solution,
- some information about the specific implementation,
- shortcomings of the solution and possible future improvements.

     
    Key changes:
    - Asynchronous Dispatch: Modified `_transfer_and_register_cpu_chunks` to
      return a list of JAX futures (unready arrays) and removed the
      `jax.block_until_ready()` stall.
    - Soft Budgeting Polling: Updated `_process_completed_saves` in the main
      feedback loop to implement a non-blocking two-stage completion check:
      1. Python Readiness: Verifies the background thread finished
         dispatching (`future.done()`).
      2. Jax Array Readiness: Performs a non-blocking hardware check
         (`array.is_ready()`) to confirm the TPU finished moving the bytes.
    - Safe Resource Lifecycle: The signal to the Scheduler to "Free" staging
      buffer slots is now strictly gatekept by cpu chunks readiness.
    - Unit test for async save + is_ready() check for the jax array stored
      in cpu backend

If the change fixes a Github issue, please include a link, e.g.,:
FIXES: #123456

# Tests

Please describe how you tested this change, and include any instructions and/or
commands to reproduce.

Manual runs of the kv host offload unit tests in v7 machine

# Checklist

Before submitting this PR, please make sure:
- I have performed a self-review of my code.
- I have necessary comments in my code, particularly in hard-to-understand areas.
- I have made or will make corresponding changes to any relevant documentation.

### CODE REVIEW

File: tpu_inference/offload/tpu_offload_connector.py [hunk 1]
```diff
@@ -2443,7 +2449,7 @@ def _async_transfer_task(req_id, *args):
                                                    num_blocks_to_save, [info],
                                                    False)
 
-                self._pending_save_futures.append((future, [info]))
+                self._pending_save_futures.append((future, [info], None))
```
Comment (juncgu-google) [inline tpu_inference/offload/tpu_offload_connector.py:2452]: do we still need to put  `_async_transfer_task` in a thread for execution? Since the copy is non-blocking, we can directly call the fuction and append the new future (a tuple of unready_chunks and info) into the `self._pending_save_futures`

Comment (saikat-royc) [reply tpu_inference/offload/tpu_offload_connector.py:2452]: addressed in https://github.com/vllm-project/tpu-inference/pull/2102

**Comment by saikat-royc** [comment]
please review https://github.com/vllm-project/tpu-inference/pull/2102
